spec/services/subscriptions/free_trial_billing_service_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Subscriptions::FreeTrialBillingService, type: :service do
+  subject(:service) { described_class.new(timestamp:) }
+
+  describe '.ending_trial_subscriptions' do
+    let(:timestamp) { Time.zone.now }
+    let(:plan) { create(:plan, trial_period: 10, pay_in_advance: true) }
+
+    context 'without any ending trial subscriptions' do
+      it do
+        create(:subscription, plan:, started_at: 2.days.ago)
+        expect(service.send(:ending_trial_subscriptions)).to be_empty
+      end
+    end
+
+    context 'when there are ending trial subscriptions' do
+      it do
+        ending_trial_subscription = create(:subscription, plan:, started_at: 10.days.ago)
+        expect(service.send(:ending_trial_subscriptions)).to eq [ending_trial_subscription]
+      end
+    end
+
+    context 'with multiple subscriptions with the same external_id' do
+      it do
+        customer = create(:customer)
+        create(:organization, customers: [customer])
+        attr = { customer:, plan:, external_id: 'abc123'}
+        new_subscription = create(:subscription, started_at: 6.days.ago, **attr)
+        _previous_subscription = create(:subscription, started_at: 10.days.ago, terminated_at: 6.days.ago, status: :terminated, **attr)
+
+        expect(service.send(:ending_trial_subscriptions)).to eq [new_subscription]
```

Comment:
Style/Send: Prefer `Object#__send__` or `Object#public_send` to `send`. (https://rubystyle.guide#prefer-public-send)